lib/cocoapods-downloader/git.rb

```diff
@@ -27,16 +27,35 @@ def self.preprocess_options(options)
                    options[:git],
                    options[:branch]]
         output = Git.execute_command('git', command)
-        match = /^([a-z0-9]*)\t.*/.match(output)
+        match = commit_from_ls_remote output, options[:branch]
 
         return options if match.nil?
 
-        options[:commit] = match[1]
+        options[:commit] = match
         options.delete(:branch)
 
         options
       end
 
+      # Matches a commit from the brances reported by git ls-remote.
+      #
+      # @note   When there is a branch and tag with the same name, it will match
+      #         the branch, since `refs/heads` is sorted before `refs/tags`.
+      #
+      # @param  [String] output
+      #         The output from git ls-remote.
+      #
+      # @param  [String] branch_name
+      #         The desired branch to match a commit to.
```

Comment:
can you add a `@return [Type]` specifying what this returns? looks like it's a string